source/computer/linux101.rst

```diff
@@ -61,18 +61,282 @@ Linux 入门
+   =============== ===================================================== ============
+   ~               匹配当前用户的家目录                                     例如 /home/seismo-learn
+   ?	           匹配任意单个字符（不能匹配零个字符）                       ?.txt 匹配 a.txt、b.txt 等文件名
+   \*	           匹配任意数量字符（能匹配零个字符）                         \*.txt 匹配 a.txt、b.txt、ab.txt 等文件名
+   [...]           匹配方括号中的任意一个字符                               [abc].txt 匹配 a.txt、b.txt 以及 c.txt 文件名
+   [start-end]     方括号扩展的简写模式，匹配一个连续的范围                   [a-z] 表示所有小写字母，[0-9] 等同于 [0123456789]
+   [^...]          匹配不在方括号里面的任意一个字符                          [^ab]c.txt 匹配 cc.txt、dc.txt 等文件名
+   [!...]          与 [^...] 等价                                         [!ab]c.txt 匹配 cc.txt、dc.txt 等文件名
+   {...}           扩展大括号里面的所有值，各个值之间使用逗号分隔              s{e,i,s}m 扩展成 sem sim ssm
+   {start..end}    大括号扩展的简写模式，扩展一个连续的范围                   s{a..c}m 扩展成 sam sbm scm
+   =============== ===================================================== ============
+
+
+匹配任意单个字符::
+
+    # 若当前目录下存在文件 a.txt 和 b.txt
+    $ ls ?.txt
+    a.txt b.txt
+    # 若当前目录下存在文件 a.txt、b.txt 和 ab.txt
+    $ ls ??.txt
+    ab.txt
+
+匹配任意数量字符::
+
+    # 若当前目录下存在文件 a.txt、b.txt 和 ab.txt
+    $ ls *.txt
+    a.txt b.txt ab.txt
+    # 若当前目录下存在文件 a.txt、b.txt 和 ab.txt
+    $ ls a*.txt
+    a.txt ab.txt
+    $ ls *b*
+    b.txt ab.txt
+
+使用方括号匹配字符::
+
+    # 若当前目录下存在文件 a.txt 和 b.txt
+    $ ls [ab].txt
+    a.txt b.txt
+    # 若当前目录下只存在文件 a.txt
+    $ ls [ab].txt
+    a.txt
+
+    # 若当前目录下存在文件 a.txt b.txt c.txt
+    $ ls [a-c].txt
+    a.txt b.txt c.txt
+
+    # 若当前目录下存在文件 aaa.txt bbb.txt aba.txt
+    $ ls ?[^a]?.txt
+    aba.txt bbb.txt
+
+使用大括号匹配字符（大括号内部的逗号前后不能有空格）::
+
+    $ echo s{e,i,s}m
+    sem sim ssm
+    # 逗号前面可以没有值，表示扩展的第一项为空
+    $ echo seismo-learn{,.md,.txt}
+    seismo-learn seismo-learn.md seismo-learn.txt
+
+    $ echo d{a..d}g
+    dag dbg dcg ddg
+
+    $ echo {3..1}
+    3 2 1
+
+.. warning::
+
+   其他 Shell （如 Zsh, csh, ksh）的扩展语法可能稍微不同，以上示例可能不适用。
 
 正则表达式
 ^^^^^^^^^^
 
+正则表达式（regular expression）描述了一种字符串匹配的模式，可以用来检查一个字符串
+是否含有某种子串、将匹配的子串做替换或者从某个字符串中取出符合某个条件的子串等。
+本文只介绍正则表达式的一些常见用法，可以参考
+`正则表达式 <https://www.bookstack.cn/read/bash-tutorial/docs-archives-regex.md>`__
+学习详细用法。也可以通过 https://regex101.com/ 在线检查正则表达式。
+
+======== ============================================================= ============
+特殊字符	     作用                                                    例子
+======== ============================================================= ============
+[]	     表示匹配的字符集合                                              [0-9],[abcde]
+()	     标记子表达式起止位置
+\*	     匹配前面的子表达式零或多次
+\+	     匹配前面的子表达式一或多次
+?	     匹配前面的子表达式零或一次
+\\	     转义字符
+.	     匹配除 \n 外的任意单个字符
+{}	     标记限定符表达式的起止。                                         {n} 表示匹配前一子表达式 n 次；{n,} 匹配至少 n 次；{n,m} 匹配 n 至 m 次
+\|	     表明前后两项二选一
+$	     匹配字符串的结尾
+^	     匹配字符串的开头；
+======== ============================================================= ============
+
+在方括号表达式中表示不接受该方括号表达式中的字符集合
+
```

Comment:
Let's remove this part. It's rarely used although useful

Reply:
okay

Reply:
Done in 87b28d2.